sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/UInt128.java

```diff
@@ -3,12 +3,55 @@
 
 package com.azure.cosmos.implementation.routing;
 
-class UInt128 {
-    long low;
-    long high;
+import java.nio.ByteBuffer;
+import java.util.Objects;
 
-    UInt128(long x, long y) {
+public class UInt128 {
+    private static final int SIZE  = Long.SIZE + Long.SIZE;
+    public static final int BYTES = SIZE / Byte.SIZE;
+    final long low;
+    final long high;
+
+    public UInt128(long x, long y) {
         this.low = x;
         this.high = y;
     }
+
+    public UInt128 (ByteBuffer byteBuffer) {
+        byteBuffer.rewind();
+        this.low = byteBuffer.getLong();
+        this.high = byteBuffer.getLong();
+    }
+
+    @Override
+    public boolean equals(final Object other) {
+        if (other instanceof UInt128) {
```

Comment:
usually smaller cases are handled first:

```
if(other not instanceof UInt128)
{
    return false;
}

return this.low == ((UInt128) other).low && this.high == ((UInt128) other).high;
```

Reply:
I see similar style being used in java boxed types like Integer